src/testRunner/unittests/tsserverProjectSystem.ts

```diff
@@ -330,19 +330,46 @@ namespace ts.projectSystem {
         return new TestSession({ ...sessionOptions, ...opts });
     }
 
-    function createSessionWithEventTracking<T extends server.ProjectServiceEvent, U extends server.ProjectServiceEvent = T>(host: server.ServerHost, eventName: T["eventName"], eventName2?: U["eventName"]) {
-        const events: (T | U)[] = [];
+    function createSessionWithEventTracking<T extends server.ProjectServiceEvent>(host: server.ServerHost, eventName: T["eventName"], ...eventNames: T["eventName"][]) {
+        const events: T[] = [];
         const session = createSession(host, {
             eventHandler: e => {
-                if (e.eventName === eventName || (eventName2 && e.eventName === eventName2)) {
-                    events.push(e as T | U);
+                if (e.eventName === eventName || eventNames.some(eventName => e.eventName === eventName)) {
+                    events.push(e as T);
                 }
             }
         });
 
         return { session, events };
     }
 
+    function createSessionWithDefaultEventHandler<T extends protocol.AnyEvent>(host: TestServerHost, eventName: T["event"], opts: Partial<server.SessionOptions> = {}, ...eventNames: T["event"][]) {
+        const session = createSession(host, { canUseEvents: true, ...opts });
+
+        return {
+            session,
+            getEvents,
+            clearEvents
+        };
+
+        function getEvents() {
+            const outputEventRegex = /Content\-Length: [\d]+\r\n\r\n/;
+            return filter(

```

Comment:
I usually prefer `mapDefined` over `filter(map(*))`, so you don't need a cast at the end and avoid an extra array.